src/Feldspar/Core/Language.hs

```diff
@@ -444,14 +444,82 @@ err msg = assertMsg msg false undef
 -- FFI.hs
 --------------------------------------------------
 
-{- Omitted
-foreignImport :: ( Type (DenResult a)
-                 , SyntacticN c b
-                 , ApplySym a b FeldDomain
-                 )
-              => String -> Denotation a -> c
-foreignImport name f = sugarSymF (ForeignImport name f)
--}
+-- | Arity specific functions for FFI
+
+foreignImport0 :: (Syntactic r, Type (Internal r))
+               => String -> Internal r -> r
+foreignImport0 s f = sugarSym0 (ForeignImport s (EqBox f))
+
+foreignImport1 :: (Syntactic a, Syntactic r, Type (Internal r), Typeable (Internal a))
+               => String -> (Internal a -> Internal r) -> a -> r
+foreignImport1 s f = sugarSym1 (ForeignImport s (EqBox f))
+
+foreignImport2 :: (Syntactic a, Syntactic b, Syntactic r, Type (Internal r),
```

Comment:
I don't think we want to link against foreign code in our test suite since it's not  portable, but having at least a gold test would be nice. Since there's no linking that checks for missing symbols we can just invent a representative foreign function, `sendInt` for example.

Reply:
Seems reasonable.